File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItOrderActionParcelable.java
package it.trade.android.sdk.model;

import android.os.Parcel;
import android.os.Parcelable;

import it.trade.android.sdk.enums.TradeItOrderAction;

public class TradeItOrderActionParcelable implements Parcelable {
    private TradeItOrderAction action;
    private String displayLabel;

    TradeItOrderActionParcelable(TradeItOrderAction action, String displayLabel) {
        this.action = action;
        this.displayLabel = displayLabel;
    }

    public TradeItOrderAction getAction() {
        return action;
    }

    public void setAction(TradeItOrderAction action) {
        this.action = action;
    }

    public String getDisplayLabel() {
        return displayLabel;
    }

    public void setDisplayLabel(String displayLabel) {
        this.displayLabel = displayLabel;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TradeItOrderActionParcelable that = (TradeItOrderActionParcelable) o;

        if (action != that.action) return false;
        return displayLabel != null ? displayLabel.equals(that.displayLabel) : that.displayLabel == null;
    }

    @Override
    public int hashCode() {
        int result = action != null ? action.hashCode() : 0;
        result = 31 * result + (displayLabel != null ? displayLabel.hashCode() : 0);
        return result;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.action == null ? -1 : this.action.ordinal());
        dest.writeString(this.displayLabel);
    }

    protected TradeItOrderActionParcelable(Parcel in) {
        int tmpAction = in.readInt();
        this.action = tmpAction == -1 ? null : TradeItOrderAction.values()[tmpAction];
        this.displayLabel = in.readString();
    }

    public static final Parcelable.Creator<TradeItOrderActionParcelable> CREATOR = new Parcelable.Creator<TradeItOrderActionParcelable>() {
        @Override
        public TradeItOrderActionParcelable createFromParcel(Parcel source) {
            return new TradeItOrderActionParcelable(source);
        }

        @Override
        public TradeItOrderActionParcelable[] newArray(int size) {
            return new TradeItOrderActionParcelable[size];
        }
    };
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/model/TradeItOrderActionParcelable.kt
package it.trade.android.sdk.model

import android.os.Parcel
import android.os.Parcelable

import it.trade.android.sdk.enums.TradeItOrderAction
import it.trade.android.sdk.enums.TradeItOrderQuantityType

class TradeItOrderActionParcelable : Parcelable {
    var action: TradeItOrderAction
    var displayLabel: String
    var supportedOrderQuantityTypes: List<TradeItOrderQuantityType>

    internal constructor(
        action: TradeItOrderAction,
        displayLabel: String,
        supportedOrderQuantityTypes: List<TradeItOrderQuantityType>
    ) {
        this.action = action
        this.displayLabel = displayLabel
        this.supportedOrderQuantityTypes = supportedOrderQuantityTypes
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as TradeItOrderActionParcelable

        if (action != other.action) return false
        if (displayLabel != other.displayLabel) return false
        if (supportedOrderQuantityTypes != other.supportedOrderQuantityTypes) return false

        return true
    }

    override fun hashCode(): Int {
        var result = action.hashCode()
        result = 31 * result + displayLabel.hashCode()
        result = 31 * result + supportedOrderQuantityTypes.hashCode()
        return result
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(this.action.ordinal)
        dest.writeString(this.displayLabel)
        dest.writeList(this.supportedOrderQuantityTypes)
    }

    protected constructor(`in`: Parcel) {
        val tmpAction = `in`.readInt()
        this.action = TradeItOrderAction.values()[tmpAction]
        this.displayLabel = `in`.readString()
        this.supportedOrderQuantityTypes = ArrayList()
        `in`.readList(this.supportedOrderQuantityTypes, TradeItOrderQuantityType::class.java.getClassLoader())
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<TradeItOrderActionParcelable> = object : Parcelable.Creator<TradeItOrderActionParcelable> {
            override fun createFromParcel(source: Parcel): TradeItOrderActionParcelable {
                return TradeItOrderActionParcelable(source)
            }

            override fun newArray(size: Int): Array<TradeItOrderActionParcelable?> {
                return arrayOfNulls(size)
            }
        }
    }
}
